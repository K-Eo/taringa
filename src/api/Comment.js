import APIBase from './APIBase'

class Comments extends APIBase {
  byId(id, params = {}) {
    return super.execute(`/comment/${id}`, params)
  }
}

export default Comments
