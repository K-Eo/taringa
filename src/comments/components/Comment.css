.Comment:last-child {
  margin-bottom: 0 !important;
}

.Comment-avatar {
  max-width: 40px;
  max-height: 40px;
  min-width: 40px;
  min-height: 40px;
}

.Comment-meta-created {
  color: #a6a6a6;
  font-size: .85rem;
  font-weight: normal;
  margin-left: 10px;
}

.Comment-body img {
  max-width: 100%;
}

.Comment iframe {
  max-width: 100%;
}

.Comment-body {
  word-break: break-word;
}

.Comment-body span[style^='position:relative'] {
  overflow: hidden;
  height: 16px;
}

.Comment-replies-toggler {
  margin-top: 10px;
  padding: 0;
  font-weight: bold;
  text-decoration: none;
}

.Comment-replies-toggler .fa {
  margin-left: 4px;
}

.Comment-replies-toggler:hover,
.Comment-replies-toggler:focus,
.Comment-replies-toggler:active {
  text-decoration: none;
}
