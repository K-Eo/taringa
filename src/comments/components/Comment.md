Basic Comment

```js
const comment = require('./comment.data.js').comment;
<Comment {...comment}/>
```
