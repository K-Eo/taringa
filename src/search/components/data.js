export const user = {
  id: '27s73',
  created: 1513920502005,
  type: 'user',
  tid: 924744,
  username: 'supr159',
  firstname: 'Diego',
  lastname: 'Lopez',
  birthday: '1992-1-18',
  avatar:
    'https://a21.t26.net/taringa/avatares/4/6/6/F/D/5/supr159/120x120_B7B.jpg',
  site: '',
  gender: 'm',
  country: 'MX',
  message: '',
  subscribedToDefaultChannels: true,
  suspended: false,
  stories: 11,
  followers: 32,
  following: 0,
  rank: 'User',
  latestStory: '2018-01-17T03:07:26.000Z',
  state: {
    following: false,
  },
}
