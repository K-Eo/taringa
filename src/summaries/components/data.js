export const image = {
  id: 'zbmsb',
  created: 1513920502005,
  edited: null,
  title: '',
  nsfw: false,
  classic: {
    tid: 'cH3t9',
    type: 'shout',
  },
  tags: ['image', 'classic'],
  slug: 'si-no-das-like-no-tendras-pelirojas-en-100-anos_zbmsb',
  type: 'story:summary',
  summary: {
    link: null,
    images: {
      slice: [
        {
          type: 'image',
          url: 'https://k50.kn3.net/taringa/2/7/9/1/4/3/20/fraudier/D4C.jpg',
          width: 779,
          height: 1026,
          format: 'jpeg',
        },
      ],
      amount: 1,
    },
    excerpt: 'Si no das like no tendras pelirojas en 100 años',
    textLength: 47,
  },
  channel: {
    type: 'channel:summary',
    id: 'ixoir',
    created: '2016-08-28T21:54:19.000Z',
    name: 'user-Fraudier',
    title: 'user-Fraudier',
    channelType: 'user_feed',
    thumbnail: '',
    background: '',
    description: '',
    category: 'unknown/unknown',
    owner: {
      id: 'ixoiq',
      created: '2016-08-28T21:54:19.000Z',
      type: 'user',
      tid: 27914320,
      username: 'Fraudier',
      firstname: 'Fraudier',
      lastname: 'ㅤ',
      birthday: '1910-9-19',
      avatar:
        'https://a12.t26.net/taringa/avatares/5/E/3/6/4/7/Fraudier/120x120_493.jpg',
      site: 'http://www.fraudierdrogbaa.com',
      gender: 'm',
      country: 'VG',
      message: 'El soldado de fermas',
      migrated: [],
      subscribedToDefaultChannels: false,
    },
    subscribers: 0,
    stories: 1120,
    latestStory: '2018-02-06T04:19:50.000Z',
    state: {
      subscription: 'unsubscribed',
    },
  },
  owner: {
    id: 'ixoiq',
    username: 'Fraudier',
    type: 'user:summary',
    firstname: 'Fraudier',
    lastname: 'ㅤ',
    avatar:
      'https://a12.t26.net/taringa/avatares/5/E/3/6/4/7/Fraudier/120x120_493.jpg',
    gender: 'm',
    country: 'VG',
  },
  upvotes: 2905,
  downvotes: 0,
  shares: 1,
  comments: 78,
  state: {},
  brandSafe: false,
  listViewSafe: false,
}

export const text = {
  id: 'zcvzz',
  created: 1513920502005,
  edited: null,
  title: '',
  nsfw: false,
  classic: {
    tid: 'cHDmR',
    type: 'shout',
  },
  tags: ['text', 'classic'],
  slug:
    'el-de-kapanga-echo-a-un-pibe-del-recital-por-tener-una-remera-con-la-b_zcvzz',
  type: 'story:summary',
  summary: {
    link: null,
    images: {
      slice: [],
      amount: 0,
    },
    excerpt:
      'El de Kapanga echó a un pibe del recital por tener una remera con la bandera inglesa, temiendo que pudiera incitar a su público a...',
    textLength: 167,
  },
  channel: {
    type: 'channel:summary',
    id: 'fswgi',
    created: '2016-08-17T22:02:01.000Z',
    name: 'user-Fixie-',
    title: 'user-Fixie-',
    channelType: 'user_feed',
    thumbnail: '',
    background: '',
    description: '',
    category: 'unknown/unknown',
    owner: {
      id: 'fswgh',
      created: '2016-08-17T22:02:01.000Z',
      type: 'user',
      tid: 24379422,
      username: 'Fixie-',
      firstname: 'Idiota',
      lastname: '.',
      birthday: '1998-4-4',
      avatar:
        'https://a21.t26.net/taringa/avatares/9/E/2/C/1/5/Fixie-/120x120_A50.jpg',
      site: 'http://untitled-pibe.tumblr.com/',
      gender: 'm',
      country: 'IT',
      message: 'zisis a bath tub',
      migrated: [],
      subscribedToDefaultChannels: false,
    },
    subscribers: 0,
    stories: 5978,
    latestStory: '2018-02-07T22:04:49.000Z',
    state: {
      subscription: 'unsubscribed',
    },
  },
  owner: {
    id: 'fswgh',
    username: 'Fixie-',
    type: 'user:summary',
    firstname: 'Idiota',
    lastname: '.',
    avatar:
      'https://a21.t26.net/taringa/avatares/9/E/2/C/1/5/Fixie-/120x120_A50.jpg',
    gender: 'm',
    country: 'IT',
  },
  upvotes: 41,
  downvotes: 0,
  shares: 0,
  comments: 0,
  state: {},
  brandSafe: false,
  listViewSafe: false,
}

export const article = {
  id: 'zcwgr',
  created: 1513920502005,
  edited: '2018-02-07T21:39:24.723Z',
  title: 'consejos para sobrevivir al ataque de un animal salvaje',
  nsfw: false,
  classic: {
    tid: 20125784,
    type: 'post',
    images: {
      '1:1': {
        type: 'image',
        url: 'https://t3.kn3.net/taringa/4/B/B/5/4/A/arcangelg80/01A.jpg',
        width: 526,
        height: 526,
        format: 'jpeg',
      },
      '4:3': {
        type: 'image',
        url: 'https://t1.kn3.net/taringa/F/0/C/8/7/C/arcangelg80/42E.jpg',
        width: 526,
        height: 395,
        format: 'jpeg',
      },
    },
  },
  tags: ['article', 'classic'],
  slug: 'consejos-para-sobrevivir-al-ataque-de-un-animal-salvaje_zcwgr',
  type: 'story:summary',
  summary: {
    link: null,
    images: {
      slice: [
        {
          type: 'image',
          url: 'https://t1.kn3.net/taringa/F/0/C/8/7/C/arcangelg80/42E.jpg',
          width: 526,
          height: 395,
          format: 'jpeg',
        },
        {
          type: 'image',
          url: 'https://k30.kn3.net/taringa/1/C/B/6/D/0/MaestroRoshiII/01B.gif',
          width: 650,
          height: 200,
          format: 'gif',
        },
        {
          type: 'image',
          url: 'https://k50.kn3.net/taringa/6/5/6/9/3/5/868.gif',
          width: 435,
          height: 250,
          format: 'gif',
        },
      ],
      amount: 10,
    },
    excerpt:
      'https://kn3.net/MaestroRoshiII/30-1-C-B-6-D-0-01B-GIF.html\nSi bien es cierto que es muy raro ser atacado por animales salvajes, por otro ...',
    textLength: 3926,
  },
  channel: {
    type: 'channel:summary',
    id: '22e',
    created: '2016-08-05T17:55:08.000Z',
    name: 'hazlo_tu_mismo',
    title: 'Hazlo tú mismo',
    channelType: 'public',
    thumbnail: 'https://k60.kn3.net/taringa/E/5/8/7/7/B/FB6.png',
    background: 'https://k60.kn3.net/taringa/F/1/F/6/F/8/E8D.png',
    description: 'Hazlo tú mismo',
    category: 'diy/tutorials',
    owner: {
      id: '17t',
      created: '2016-07-29T16:57:34.000Z',
      type: 'user',
      tid: 2987,
      username: 'Taringa',
      firstname: 'Taringa!',
      lastname: 'Inteligencia Colectiva',
      birthday: '1994-4-14',
      avatar:
        'https://a11.t26.net/taringa/avatares/3/D/6/2/3/8/Taringa/120x120_943.jpg',
      site: 'http://www.taringa.net',
      gender: 'm',
      country: 'AR',
      message: 'Inteligencia Colectiva',
      migrated: ['followers', 'classicCategories'],
      subscribedToDefaultChannels: true,
    },
    subscribers: 319050,
    stories: 45398,
    latestStory: '2018-02-07T22:35:28.568Z',
    state: {
      subscription: 'unsubscribed',
    },
  },
  owner: {
    id: 'a7716',
    username: 'arcangelg80',
    type: 'user:summary',
    firstname: 'angel',
    lastname: 'gonzalez',
    avatar:
      'https://a21.t26.net/taringa/avatares/A/4/6/B/2/0/arcangelg80/120x120_CDD.jpg',
    gender: 'm',
    country: 'VE',
  },
  upvotes: 9,
  downvotes: 0,
  shares: 0,
  comments: 14,
  state: {},
  brandSafe: true,
  listViewSafe: false,
}

export const link = {
  id: 'zcxyl',
  created: 1513920502005,
  edited: null,
  title: 'Godoy Cruz-Lanús, Superliga: juegan en Mendoza un partido pendiente',
  nsfw: false,
  tags: ['link', 'brainiac'],
  slug:
    'godoy-cruz-lanus-superliga-juegan-en-mendoza-un-partido-pendiente_zcxyl',
  type: 'story:summary',
  summary: {
    link: {
      type: 'link',
      url: 'https://www.lanacion.com.ar/2107299-godoy-cruz-lanus-superliga',
      resourceType: 'html',
      originalUrl:
        'http://www.lanacion.com.ar/2107299-godoy-cruz-lanus-superliga?utm_source=taringa&utm_campaign=rss',
      provider: {
        type: 'provider',
        url: 'https://www.lanacion.com.ar',
        name: 'Lanacion',
        display: 'www.lanacion.com.ar',
      },
      favicon: 'https://k50.kn3.net/taringa/F/4/C/5/E/8/3D0.png',
      title:
        'Godoy Cruz-Lanús, Superliga: juegan en Mendoza un partido pendiente',
      description:
        "Godoy Cruz y Lanús juegan hoy en Mendoza el partido que tienen pendiente de la décima fecha de la Superliga del fútbol argentino, postergado por la participación del 'Granate´ en la final de la Copa Libertadores 2017.",
      images: [
        {
          type: 'image',
          url: 'https://k50.kn3.net/taringa/9/F/2/5/E/D/EB2.jpg',
          width: 900,
          height: 599,
          format: 'jpeg',
        },
      ],
      media: null,
      authors: [],
    },
    images: {
      slice: [],
      amount: 0,
    },
    excerpt: '',
    textLength: 0,
  },
  channel: {
    type: 'channel:summary',
    id: '1jq',
    created: '2016-08-04T21:06:31.000Z',
    name: 'deportes',
    title: 'Deportes',
    channelType: 'public',
    thumbnail: 'https://k60.kn3.net/taringa/9/E/E/B/3/0/589.png',
    background: 'https://k60.kn3.net/taringa/C/7/E/1/C/A/E4C.jpg',
    description: 'Deportes',
    category: 'sports/other',
    owner: {
      id: '17t',
      created: '2016-07-29T16:57:34.000Z',
      type: 'user',
      tid: 2987,
      username: 'Taringa',
      firstname: 'Taringa!',
      lastname: 'Inteligencia Colectiva',
      birthday: '1994-4-14',
      avatar:
        'https://a11.t26.net/taringa/avatares/3/D/6/2/3/8/Taringa/120x120_943.jpg',
      site: 'http://www.taringa.net',
      gender: 'm',
      country: 'AR',
      message: 'Inteligencia Colectiva',
      migrated: ['followers', 'classicCategories'],
      subscribedToDefaultChannels: true,
    },
    subscribers: 317511,
    stories: 275050,
    latestStory: '2018-02-07T23:25:13.000Z',
    state: {
      subscription: 'unsubscribed',
    },
  },
  owner: {
    id: '1kphe',
    username: 'lanacion',
    type: 'user:summary',
    firstname: 'La Nación',
    lastname: 'RSS',
    avatar:
      'https://a21.t26.net/taringa/avatares/D/0/8/9/1/4/lanacion/120x120_96C.jpg',
    gender: 'm',
    country: 'AR',
  },
  upvotes: 0,
  downvotes: 0,
  shares: 0,
  comments: 0,
  state: {},
  brandSafe: false,
  listViewSafe: false,
}
