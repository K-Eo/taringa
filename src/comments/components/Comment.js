import React, { Component } from 'react'
import TimeAgo from 'react-timeago'
import { Link } from 'react-router-dom'

import './Comment.css'
import Action from '../../common/Action.js'
import { esFormatter } from '../../Utils'

class Comment extends Component {
  constructor(props) {
    super(props)
    this.handleOnClik = this.handleOnClik.bind(this)
    this.state = {
      showReplies: props.showReplies || false,
    }
  }

  handleOnClik() {
    window.scrollTo(0, 0)
  }

  toggleRepliesVisibility(e) {
    e.preventDefault()
    this.setState({ showReplies: !this.state.showReplies })
  }

  renderRepliesToggler() {
    const { showReplies } = this.state
    const { replies } = this.props

    if (replies && replies.items && replies.items.length > 0) {
      const repliesCount = replies.items.length

      return (
        <button
          className="btn btn-link text-dark Comment-replies-toggler mb-2"
          onClick={this.toggleRepliesVisibility.bind(this)}
        >
          {showReplies ? (
            <span>
              {`Ocultar ${repliesCount.pluralize('respuesta', null, true)} `}
              <i className="fa fa-chevron-up" />
            </span>
          ) : (
            <span>
              {`Ver ${repliesCount.pluralize(
                'respuesta',
                null,
                repliesCount === 1
              )} `}
              <i className="fa fa-chevron-down" />
            </span>
          )}
        </button>
      )
    }

    return null
  }

  render() {
    const { showReplies } = this.state
    const { body, created, downvotes, owner, replies, upvotes } = this.props

    return (
      <div className="Comment mb-2 d-flex">
        <div className="Comment-avatar mr-3 mr-lg-4">
          <img
            alt={owner.username}
            className="rounded Comment-avatar"
            src={owner.avatar}
          />
        </div>
        <div className="Comment-content">
          <h6 className="Comment-author">
            <Link
              className="text-dark"
              onClick={this.handleOnClik}
              to={`/u/${owner.username}`}
            >
              {owner.username}
            </Link>

            <TimeAgo
              className="Comment-meta-created"
              date={created}
              formatter={esFormatter}
            />
          </h6>

          <div
            className="Comment-body"
            dangerouslySetInnerHTML={{ __html: body }}
          />

          <p className="my-1">
            <Action className="mr-3" icon="far fa-thumbs-up">
              {upvotes.humanize()}
            </Action>

            <Action className="mr-3" icon="far fa-thumbs-down">
              {downvotes.humanize()}
            </Action>

            <Action className="d-none d-sm-inline">RESPONDER</Action>
            <Action className="d-sm-none" icon="far fa-comment-alt" />
          </p>

          {this.renderRepliesToggler()}

          {showReplies && (
            <div className="Comment-replies mt-2">
              {replies.items.map(r => <Comment key={r.id} {...r} />)}
            </div>
          )}
        </div>
      </div>
    )
  }
}

export default Comment
