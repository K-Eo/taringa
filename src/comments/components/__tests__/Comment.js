import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Comment from '../Comment'
import { comment, commentWithReply, commentWithReplies } from '../comment.data'

describe('Comment', () => {
  beforeEach(() => {
    Date.now = jest.fn(() => 1513924592705)
  })

  it('exists', () => {
    expect(Comment).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<Comment />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders with replies', () => {
    const wrapper = shallow(<Comment id="foo" />)

    expect(wrapper).toMatchSnapshot()
  })
})
