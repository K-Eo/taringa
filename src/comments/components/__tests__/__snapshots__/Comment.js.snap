// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment renders default 1`] = `
<Commentable
  body="#BODY"
  created={0}
  downvotes={0}
  owner="#OWNER"
  upvotes={0}
/>
`;

exports[`Comment renders with replies 1`] = `
<Commentable
  body="#BODY"
  comment={
    Object {
      "replies": "foo",
    }
  }
  created={0}
  downvotes={0}
  owner="#OWNER"
  upvotes={0}
/>
`;
